vidur/config/config.py

```diff
@@ -1,109 +1,750 @@
+        default=0.02,
+        metadata={"help": "NCCL CPU launch overhead in ms."},
+    )
+    nccl_cpu_skew_overhead_per_device_ms: float = field(
+        default=0.0,
+        metadata={"help": "NCCL CPU skew overhead per device in ms."},
+    )
+    num_training_job_threads: int = field(
+        default=-1,
+        metadata={"help": "Number of training job threads."},
+    )
+    skip_cpu_overhead_modeling: bool = field(
+        default=True,
+        metadata={"help": "Whether to skip CPU overhead modeling."},
+    )
+
+
+@dataclass
+class LinearRegressionExecutionTimePredictorConfig(BaseExecutionTimePredictorConfig):
+    polynomial_degree: List[int] = field(
+        default_factory=lambda: list(range(1, 6)),
+        metadata={"help": "Polynomial degree for linear regression."},
+    )
+    polynomial_include_bias: List[bool] = field(
+        default_factory=lambda: [True, False],
+        metadata={"help": "Polynomial include bias for linear regression."},
+    )
+    polynomial_interaction_only: List[bool] = field(
+        default_factory=lambda: [True, False],
+        metadata={"help": "Polynomial interaction only for linear regression."},
+    )
+    fit_intercept: List[bool] = field(
+        default_factory=lambda: [True, False],
+        metadata={"help": "Fit intercept for linear regression."},
+    )
+
+    @staticmethod
+    def get_type():
+        return ExecutionTimePredictorType.LINEAR_REGRESSION
+
+
+@dataclass
+class RandomForrestExecutionTimePredictorConfig(BaseExecutionTimePredictorConfig):
+    num_estimators: List[int] = field(
+        default_factory=lambda: [250, 500, 750],
+        metadata={"help": "Number of estimators for random forest."},
+    )
+    max_depth: List[int] = field(
+        default_factory=lambda: [8, 16, 32],
+        metadata={"help": "Maximum depth for random forest."},
+    )
+    min_samples_split: List[int] = field(
+        default_factory=lambda: [2, 5, 10],
+        metadata={"help": "Minimum samples split for random forest."},
+    )
+
+    @staticmethod
+    def get_type():
+        return ExecutionTimePredictorType.RANDOM_FORREST
+
+
+@dataclass
+class ClusterConfig:
+    num_replicas: int = field(
+        default=1,
+        metadata={"help": "Number of replicas."},
+    )
+    replica_config: ReplicaConfig = field(default_factory=ReplicaConfig)
+    execution_time_predictor_config: BaseExecutionTimePredictorConfig = field(
+        default_factory=RandomForrestExecutionTimePredictorConfig,
+        metadata={"help": "Execution time predictor config."},
+    )
+    global_scheduler_config: BaseGlobalSchedulerConfig = field(
+        default_factory=RoundRobinGlobalSchedulerConfig,
+        metadata={"help": "Global scheduler config."},
+    )
+    replica_scheduler_config: BaseReplicaSchedulerConfig = field(
+        default_factory=SarathiSchedulerConfig,
+        metadata={"help": "Replica scheduler config."},
+    )
+    metrics_config: MetricsConfig = field(
```

Comment:
i guess we should move the metrics config and  execution time predictor config one level up?